sdk/logs/src/main/java/io/opentelemetry/sdk/logs/data/LogDataBuilder.java

```diff
@@ -27,6 +28,7 @@
   @Nullable private String severityText;
   @Nullable private String name;
   private Body body = Body.stringBody("");
+  private Clock clock = Clock.getDefault();
```

Comment:
Is this the right place to save a copy of the clock? Why not just have the clock in the emitter/provider and have it do the work of defaulting the epoch time on the log data?

Reply:
I assumed that the zero check and defaulting to "now" was encapsulated in the `LogDataBuilder` for good reason. If we move it up, then it requires all users of the `LogDataBuilder` to go thru the `SLEProviderBuilder` and `SLEProvider` tangle.

Right now, it's possible for users to just say `LogData.builder().set(...).set(...).build()` to get a `LogData` instance. If we move the logic up, then that defaulting gets missed.

I see what you're getting at, though, and it does feel weird to have the clock defaulted in two places (here and in `SdkLogEmitterProviderBuilder`. Not sure what's the right thing to do here.

Reply:
oh, I see the issue. I forgot the static `LogData.builder()` method didn't interact with the provider/emitter. Hmm. Ugly.  I need to ponder this. The builder is a little weird here in general.

Reply:
Here's the (ok, "a" solution 😄) solution:

- Remove the `setClock(..)` method from `Logbuilder`.
- When `SdkLogEmitter#logBuilder()` is called, call `logBuilder#setEpoch(..)` with the current epoch according to the clock configured with the `SdkLogEmitterProvider`.
- The user can still override the epoch, but if not, its set to the epoch according to the clock at the time `logBuilder()` was called. And no need for `LogBuilder` to have a clock.

Reply:
@breedx-splk and I talked through this option, and he (rightfully) asserted that if we just put the epoch value in at builder creation, then the builder wouldn't be reusable (without manually setting the epoch value on every usage thereafter).

Reply:
Yeah, for max accuracy, I think the time should be obtained from the clock at the moment of pojo creation, not builder creation.

Reply:
It still feels wrong to have a clock associated with a record.

Another alternative is to take advantage of the fact that `SdkLogEmitter#SdkLogBuilder` delegates to `LogDataBuilder`, which means that we can track whether `SdkLogBuilder#setEpoch()` has been called with a boolean. If it hasn't been called when `emit()` is invoked, then we can set a default value using the clock.